include/coreinit/scheduler.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+#include <wut.h>
+
+/**
+ * \defgroup coreinit_scheduler Scheduler
+ * \ingroup coreinit
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+extern void* __OSSchedulerLock;
+
+void
+__OSEnableScheduler();
+
+void
+__OSDisableScheduler();
+
+void
+__OSLockScheduler(void* lock);
```

Comment:
void *lock kinds of implies you need to pass in a lock here, its actually a "lock id" used only to have a unique identifier for who is holding the lock so they can have some debug asserts for added safety.

maybe just rename it to lockId ? I am not bothered about it being void * or uint32_t .

Reply:
Done